src/sqlfluff/utils/functional/segment_predicates.py

```diff
@@ -29,27 +29,29 @@ def _(segment: BaseSegment):
     return _
 
 
-def is_type(*seg_type: str) -> Callable[[BaseSegment], bool]:
-    """Returns a function that determines if segment is one of the types."""
+def raw_upper_is(*raws: str) -> Callable[[BaseSegment], bool]:
+    """Returns a function that determines if segment matches one of the raw inputs."""
 
     def _(segment: BaseSegment):
-        return segment.is_type(*seg_type)
+        return segment.raw_upper in raws
 
     return _
 
 
-def is_name(*seg_name: str) -> Callable[[BaseSegment], bool]:
-    """Returns a function that determines if segment is one of the names."""
+def is_type(*seg_type: str) -> Callable[[BaseSegment], bool]:
+    """Returns a function that determines if segment is one of the types."""
 
     def _(segment: BaseSegment):
-        return segment.is_name(*seg_name)
+        return segment.is_type(*seg_type)
 
     return _
 
 
 def is_keyword(*keyword_name) -> Callable[[BaseSegment], bool]:
     """Returns a function that determines if it's a matching keyword."""
-    return and_(is_type("keyword"), is_name(*keyword_name))
+    return and_(
+        is_type("keyword"), raw_upper_is(*[raw.upper() for raw in keyword_name])
```

Comment:
Nitpick: For performance, I wonder if we'd want to `assert` that the strings in `keyword_name` are already uppercase (to avoid calling `upper()`)

Reply:
Good idea :+1:

Reply:
actually here - we're calling `.upper()` on the elements of `keyword_name`, which is going to be quite short relative to the segments. I think the performance hit here is minimal.